src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -280,6 +280,153 @@ variant_map_type _activity_classifier_prepare_data_impl(const gl_sframe &data,
     return result_dict;
 }
 
+variant_map_type _activity_classifier_prepare_data_aug_impl(const gl_sframe &data,
+                                                        const std::vector<std::string> &features,
+                                                        const std::string &session_id,
+                                                        const int &prediction_window,
+                                                        const int &predictions_in_chunk,
+                                                        const std::string &target,
+                                                        bool verbose) {
+
+    DASSERT_TRUE(features.size() > 0);
+    DASSERT_TRUE(prediction_window > 0);
+    DASSERT_TRUE(predictions_in_chunk > 0);
+    DASSERT_TRUE(data.contains_column(session_id));
+    for (TURI_ATTRIBUTE_UNUSED_NDEBUG auto &feat : features) {
```

Comment:
I would probably wraps this for-loop in a 
```
#ifndef NDEBUG
....
#endif
```
You don't want to be iterating over the features for no reason in release mode. Although perhaps the compiler would be smart enough not to let that happen.